Automattic/jetpack
# Payments Block: localize block variations

Fixes https://github.com/Automattic/jetpack/issues/37014

## Proposed changes:

In the editor, the names of the 3 block variations inside the Payments block are not localized. This PR fixes it.
<img width="400" alt="Screenshot 2024-04-22 at 6 30 56 PM" src="https://github.com/Automattic/jetpack/assets/1620183/4cd37a80-fe07-4da4-bc35-ddb4ea51a56b">

These 3 blocks are localized: when adding each block individually through the block inserter we can see their title, description, etc., in the interface language. However, in this specific instance, the block names are read directly from each block's `block.json` metadata file, and this JSON content is not localized.

Indeed, translation data from a `block.json` file is included server side in a PO file, whose data is [then provided to the front end](https://github.com/WordPress/WordPress/blob/1c0e271c2fd25796beac4b5ae1c0cea2ffbca773/wp-admin/edit-form-blocks.php#L110). We can access the localized data in `wp.blocks`.

## Jetpack product discussion


Project thread: pf5801-Aw-p2

## Does this pull request change what data or activity we track or use?


No.

## Testing instructions:






### To reproduce the issue
- Spin up a test Jetpack site
- Set the user's language to something different than English
- Create a new post
- Add the _Payments_ block
- Notice the names of the 3 blocks are in English

### To test the fix
- Checkout this branch
- Repeat the step above
- Notice the names are now translated
<img width="400" alt="Screenshot 2024-05-31 at 11 15 48 AM" src="https://github.com/Automattic/jetpack/assets/1620183/c9937c3d-f9f6-4896-8ea6-35f6198aec65">
